src/Macro.php

```diff
@@ -63,7 +63,11 @@ protected function initPhpDoc($method)
 
     protected function addLocationToPhpDoc()
     {
-        $enclosingClass = $this->method->getClosureScopeClass();
+        if ($this->method->name === '__invoke') {
```

Comment:
Given a macro accepts `object|callable`, possible values are:
* `callable` - `function () { }`
* `callable` - `[$instance, 'method']`
* `object` - `new Foo()`

This should be `$enclosingClass === null` so it covers cases all cases where `$this->method` is not an anonymous function.

Reply:
Feel free to send a PR!